summarization/main.py

```diff
@@ -89,75 +90,51 @@ def main():
     logger.info(f"Starting Summarization Service (version {__version__})")
     
     try:
-        # Load configuration from environment
-        message_bus_type = os.getenv("MESSAGE_BUS_TYPE", "rabbitmq")
-        message_bus_host = os.getenv("MESSAGE_BUS_HOST", "messagebus")
-        message_bus_port = int(os.getenv("MESSAGE_BUS_PORT", "5672"))
-        message_bus_user = os.getenv("MESSAGE_BUS_USER", "guest")
-        message_bus_password = os.getenv("MESSAGE_BUS_PASSWORD", "guest")
-        
-        doc_store_type = os.getenv("DOC_STORE_TYPE", "mongodb")
-        doc_store_host = os.getenv("DOC_DB_HOST", "documentdb")
-        doc_store_port = int(os.getenv("DOC_DB_PORT", "27017"))
-        doc_store_name = os.getenv("DOC_DB_NAME", "copilot")
-        doc_store_user = os.getenv("DOC_DB_USER", "")
-        doc_store_password = os.getenv("DOC_DB_PASSWORD", "")
-        
-        vector_store_type = os.getenv("VECTOR_STORE_TYPE", "inmemory")
-        vector_store_host = os.getenv("VECTOR_DB_HOST", "vectorstore")
-        vector_store_port = int(os.getenv("VECTOR_DB_PORT", "6333"))
-        vector_store_collection = os.getenv("VECTOR_DB_COLLECTION", "message_embeddings")
-        
-        # Summarization configuration
-        llm_backend = os.getenv("LLM_BACKEND", "mock")
-        llm_model = os.getenv("LLM_MODEL", "mistral")
-        top_k = int(os.getenv("TOP_K", "12"))
-        citation_count = int(os.getenv("CITATION_COUNT", "12"))
-        retry_max_attempts = int(os.getenv("RETRY_MAX_ATTEMPTS", "3"))
-        retry_backoff_seconds = int(os.getenv("RETRY_BACKOFF_SECONDS", "5"))
+        # Load configuration from schema with typed access
+        config = load_typed_config("summarization")
```

Comment:
Fixed: replaced `load_config("summarization")` with `load_typed_config("summarization")` to match the import and ensure typed attribute access. This prevents a NameError and aligns with the schema-based config pattern.